classes/class-wc-connect-shipping-label.php

```diff
@@ -36,89 +36,118 @@ protected function get_form_data( WC_Order $order ) {
 					'width' => $width
 				);
 			}
-			$formData[ 'cart' ] = $contents;
+			$form_data[ 'cart' ] = $contents;
+
+			$dest_address = $order->get_address( 'shipping' );
+			$form_data[ 'dest_name' ] = trim( $dest_address[ 'first_name' ] . ' ' . $dest_address[ 'last_name' ] );
+			$form_data[ 'dest_company' ] = $dest_address[ 'company' ];
+			$form_data[ 'dest_address_1' ] = $dest_address[ 'address_1' ];
+			$form_data[ 'dest_address_2' ] = $dest_address[ 'address_2' ];
+			$form_data[ 'dest_city' ] = $dest_address[ 'city' ];
+			$form_data[ 'dest_state' ] = $dest_address[ 'state' ];
+			$form_data[ 'dest_postcode' ] = $dest_address[ 'postcode' ];
+			$form_data[ 'dest_country' ] = $dest_address[ 'country' ];
+
+			return $form_data;
+		}
 
-			foreach( $order->get_address( 'shipping' ) as $key => $value ) {
-				$formData[ 'orig_' . $key ] = '';
-				$formData[ 'dest_' . $key ] = $value;
+		protected function get_states_map() {
+			$result = array();
+			foreach( WC()->countries->get_states() as $country => $states ) {
+				$result[ $country ] = array();
+				foreach ( $states as $code => $name ) {
+					$result[ $country ][ $code ] = html_entity_decode( $name );
+				}
 			}
+			return $result;
+		}
 
-			return $formData;
+		protected function get_countries() {
+			$result = array();
+			foreach( WC()->countries->get_countries() as $code => $name ) {
+				$result[ $code ] = html_entity_decode( $name );
+			}
+			return $result;
 		}
 
 		protected function get_form_layout() {
 			$layout = array();
 
 			$address_fields = array(
 				array(
-					'key' => 'first_name',
-					'validation_hint' => 'This field is required.',
-				),
-				array(
-					'key' => 'last_name',
+					'key' => 'name',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'company',
 				),
 				array(
 					'key' => 'address_1',
-					'validation_hint' => 'This field is required.',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'address_2',
 				),
 				array(
 					'key' => 'city',
-					'validation_hint' => 'This field is required.',
+					'validation_hint' => __( 'Required.', 'woocommerce' ),
 				),
 				array(
 					'key' => 'state',
-					'validation_hint' => 'This field is required.',
+					'type' => 'state',
+					'dataset' => $this->get_states_map(),
```

Comment:
We'll likely want to keep populating these with data from the WC instance.. so perhaps a custom `type` that renders with a specific country/state component makes the most sense.